state.c

```diff
@@ -470,6 +470,52 @@ static void pv_state_set_default_conditions(struct pv_state *s)
 	pv_state_set_default_conditions_group(s, a, NULL, "STARTED");
 }
 
+static bool pv_state_group_requires_reboot(struct pv_group *g)
+{
+	int len;
+	char *name;
+
+	if (!g || !g->name)
+		return true;
+
+	name = g->name;
+	len = strlen(name);
+
+	// some groups need reboot to keep backwards compatibility with runlevel behavior
+	if (pv_str_matches(name, len, "data", strlen("data")) ||
+	    pv_str_matches(name, len, "root", strlen("root")) ||
+	    pv_str_matches(name, len, "platform", strlen("platform"))) {
```

Comment:
imo this legacy behaviour should only apply if no groups.json has been loaded/defined. This brings us closer to the spec where our default behaviour is actually realized through an implicit groups.json

Reply:
it will be like that, but groups.json will not be added until the third drop of this series. For now, explicit restart_policy in run.json overrides whatever group the container belongs to, but we keep backwards compatibility and the policy will be set according to the default groups ('system' to data, root and platform; 'container' to app) if not defined